lighthouse-core/gather/computed/metrics/metric.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const ComputedArtifact = require('../computed-artifact');
+
+class MetricArtifact extends ComputedArtifact {
+  /** @type {string} */
+  get name() {
+    throw new Error('Unimplemented');
+  }
+
+  /**
+   * @param {LH.Gatherer.Artifact.MetricComputationData} data
+   * @param {Object} artifacts
+   * @return {Promise<LH.Gatherer.Artifact.LanternMetric>}
+   */
+  computeSimulatedMetric(data, artifacts) {
+    return artifacts[`requestLantern${this.name}`](data);
+  }
+
+  /**
+   * @param {LH.Gatherer.Artifact.MetricComputationData} data
+   * @param {Object} artifacts
+   * @return {Promise<LH.Gatherer.Artifact.Metric>}
+   */
+  computeObservedMetric(data, artifacts) { // eslint-disable-line no-unused-vars
+    throw new Error('Unimplemented');
+  }
+
+  /**
+   * @param {LH.Gatherer.Artifact.MetricComputationData} data
```

Comment:
The primary benefits:

1. All callers of `compute*` should pass in the same set of properties to ensure that caching works since it's based on deep equality (maybe we should split this into separate typedef?)
2. Achieve 1 while still allowing one-off computations for alternative settings, for example, regardless of the throttling specified, load-fast-enough-4-pwa will always use lantern to compute the 10s TTI for consistency (and enable the audit to not stick the ugly toplevel warning when called without DT throttling)
3. All artifacts would need to do the networkRecords traceOfTab as their first two lines